src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,187 @@
+        compatibility=[interpreter_constraints] if interpreter_constraints else None,
+      )
+    )
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
+  def test_respects_config_file(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], config="[pylint]\ndisable = C0114\n")
+    assert result.exit_code == 0
+    assert result.stdout.strip() == ""
+
+  def test_respects_passthrough_args(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="target")
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], passthrough_args="--disable=C0114")
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_transitive_dep(self) -> None:
+    self.create_library(path=self.source_root, target_type="python_library", name="library")
+    self.create_library(
+      path=self.source_root, target_type="python_library", name="transitive_dep",
+      sources=["transitive_dep.py"], dependencies=[":library"],
+    )
+    self.write_file(
+      FileContent(
+        path="transitive_dep.py",
+        content=dedent(
+          """\
+          from nonexistent import erroneous
+          VAR = 42
+          """
+        ).encode(),
+      )
+    )
+    source = FileContent(
+      path="test_transitive_dep.py",
+      content=dedent(
+        """\
+        '''Docstring'''
+        from transitive_dep import VAR
+        assert VAR == 42
+        """
+      ).encode(),
+    )
+    self.create_python_test_target([source], dependencies=[":transitive_dep"])
```

Comment:
Bad copy pasta from `python_test_runner_integration_test.py`. Here, we don't care that it's a `python_tests` target. All we care about is it being a Python target. You should be able to use the proposed `create_python_library` method from the above comment.